sql/plsql/PlSqlParser.g4

```diff
@@ -6083,7 +6083,11 @@ logical_expression
     ;
 
 unary_logical_expression
-    : NOT? multiset_expression (IS NOT? logical_operation)*
+    : NOT? multiset_expression unary_logical_operation*
+    ;
+
+unary_logical_operation
+    : (IS NOT? logical_operation)
```

Comment:
Parentheses of `(IS NOT? logical_operation)` should be removed. They are unnecessary. Also, is there a test for this for the parse tree?

Reply:
Thanks, fixed.

Also, I realized that multiple unary logical operations were allowed before this PR - e.g. `foo IS NULL IS NOT NAN`, which doesn't seem right and is also rejected by an Oracle server. So I changed the `*` to a `?` in the `unary_logical_expression` rule. Does that look correct?

There was some example code containing a unary logical operation more or less implicitly, so I've added a file containing some more explicit examples.